### PR TITLE
feature: use keepalvied to add vip for virtual cluster

### LINKED ISSUE #635: support ha for tenant‘s VirtualCluster
**What would you like to be added**:

Currently, when creating a tenant's cluster, the services of the tenant's cluster cannot provide services to the outside world through a high-availability mode. Therefore, when creating a tenant cluster, kubenest also needs the ability to dynamically create keepalive plus haproxy or directly use metalLB to expose services externally.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by village-way**
#### What type of PR is this?

/kind feature

#### What does this PR do?

add vip for kube in kube for tenant cluster
#### Which issue(s) does this PR fix?

Fixes #635 
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by duanmengkk** [comment]
在deploy下面建一个目录，把keepalive的yaml传上去吧，不然代码开源不完整了

**Comment by village-way** [comment]
> 在deploy下面建一个目录，把keepalive的yaml传上去吧，不然代码开源不完整了

add keepalived manifest file in `deploy/k8s-in-k8s/manifeat/keepalived`

File: deploy/k8s-in-k8s/manifest/keepalived/keepalived-sts.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            - labelSelector:
+                matchExpressions:
+                  - key: virtualCluster-app
+                    operator: In
+                    values:
+                      - keepalived
+              topologyKey: "kubernetes.io/hostname"
+      initContainers:
+        - name: init-keepalived-config
+          image: {{ .ImageRepository }}/kubectl:v1.25.7-eki.3.0.0
```
Comment (duanmengkk) [inline deploy/k8s-in-k8s/manifest/keepalived/keepalived-sts.yaml]:
```suggestion
          image: {{ .ImageRepository }}/kubectl:v1.25.7
```

File: deploy/k8s-in-k8s/manifest/keepalived/keepalived-config.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  master: |+
+    vrrp_instance VI_1 {
+        state MASTER
+        interface __IFNAME__
+        virtual_router_id 51
+        priority 100
+        advert_int 1
+        authentication {
+            auth_type PASS
+            auth_pass Ae*vee1op@ahng5a
```
Comment (duanmengkk) [inline deploy/k8s-in-k8s/manifest/keepalived/keepalived-config.yaml]:
```suggestion
            auth_pass XXXXXX
```

File: deploy/k8s-in-k8s/manifest/keepalived/keepalived-sts.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            - "sh"
+            - "-c"
+            - "ROLE=$(kubectl get no -L kosmos.io/keepalived-role | grep ${NODE_NAME} | awk '{print $6}'); NODEIP=$(kubectl get no -o wide | grep ${NODE_NAME} | awk '{print $6}'); IFNAME=$(ip -o addr show | grep $NODEIP | awk '{print $2}'); if [ $ROLE = 'master' ]; then cp /config/master /shared/keepalived.conf; else cp /config/backup /shared/keepalived.conf; fi; sed -i 's/__IFNAME__/'$IFNAME'/g' /shared/keepalived.conf;"
+          volumeMounts:
+            - name: keepalived-conf
+              mountPath: /config
+            - name: shared
+              mountPath: /shared/
+      containers:
+        - image: {{ .ImageRepository }}/keepalived:v1.25.7-eki.3.0.0
```
Comment (duanmengkk) [inline deploy/k8s-in-k8s/manifest/keepalived/keepalived-sts.yaml]: Same as above

Comment (village-way) [reply deploy/k8s-in-k8s/manifest/keepalived/keepalived-sts.yaml]: updated

File: deploy/k8s-in-k8s/manifest/keepalived/keepalived-config.yaml (hunk 2 above)
Comment (village-way) [reply deploy/k8s-in-k8s/manifest/keepalived/keepalived-config.yaml]: replace the auth_pass

**Comment by duanmengkk** [comment]
resolve the conflicts

**Comment by village-way** [comment]
this feature already in #689 , close this pr.

**Pull request closed without merging**